### PR TITLE
Build out new ICU Ventilation Guide

### LINKED ISSUE #40: Build out the ICU ventilation screen
From the main menu there is a button which takes you to the ICU ventilation screen.  That screen is currently **really** basic.

**We need to build build out a detailed version of the ICU ventilation screen**

@greggmiller Can you please provide some information in this issue on what should be in this section?  Once that is done assign the issue to Marc Edwards.

@marcedwards Once greg has provided info, can you provide some design guidance in this issue in on how it should look.  Once this is done, can you label this issue as "Ready for Dev" so that one of the devs can pick it up.

### PR DESCRIPTION
**Pull request by ericjiang97**
Provide a more detailed summary of the changes in this PR here.  What does it do?

- [x] `release-notes.txt` has been updated.

Resolves #40 

## Changes

- Added ventilation guide
- Initial refactor of TabBarView (added usable TabBarView screen)
- Partial refactor of the guide list containers

## Screenshots

https://ejiang.d.pr/ur4Jph

## Things to note

- Refactoring work on TabBarView (there could be more refactoring work done on the cards using `reusable_card_base.dart` for work on the PPE guide.

### CODE REVIEW

**Comment by lukesleeman** [comment]
I've also assigned @sallychung to this one so she can use her amazing QA 👀 on it 😁

File: lib/models/ventilation_guide.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class VentilationGuideItem {
```
Comment (maks) [inline lib/models/ventilation_guide.dart:1]: :+1: :+1:  for keeping with modelling the data @ericjiang97

File: lib/view/view_templates/tab_view_template.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  final Color color;
+  final Color indicatorColor;
+
+  const TabViewTemplate(this.title,
+      {@required this.tabs,
+      @required this.children,
+      this.color = AppColors.green50,
+      this.actions,
+      this.indicatorColor});
```
Comment (maks) [inline lib/view/view_templates/tab_view_template.dart]: pls keep trailing comma for consistent formatting when function params span multiple lines

File: lib/view/view_templates/tab_view_template.dart [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+
+import '../../style.dart';
+
+class TabViewTemplate extends StatelessWidget {
```
Comment (maks) [inline lib/view/view_templates/tab_view_template.dart:5]: :+1:  very nice work getting duplicate code into a reusable widget

File: lib/widget/containers/ventilation_guide_container.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+
+import '../../models/ventilation_guide.dart';
+import '../../style.dart';
+import '../../widget/lists/ventilation_guide_card.dart';
+
+class VentilationGuideContainer extends StatelessWidget {
+  final List<VentiliationGuideSection> guide;
```
Comment (maks) [inline lib/widget/containers/ventilation_guide_container.dart]: would be nice to rename the `guide` prop to make it obvious its actually a list of guide sections

File: lib/widget/containers/ventilation_guide_container.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  const VentilationGuideContainer({this.guide});
+
+  List<Widget> renderList() {
```
Comment (maks) [inline lib/widget/containers/ventilation_guide_container.dart]: good to make this function private

File: lib/widget/lists/ventilation_guide_card.dart [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import '../../models/ventilation_guide.dart';
+import '../../style.dart';
+import '../../widget/cards/reusable_card_base.dart';
+import '../../widget/lists/string_list.dart';
+
+class VentilationGuideCard extends StatelessWidget {
+  final VentiliationGuideSection sectionItem;
+  final Color backgroundColor;
+
+  Widget getList(List<String> notes) {
```
Comment (maks) [inline lib/widget/lists/ventilation_guide_card.dart]: private pls

File: lib/widget/lists/ventilation_guide_card.dart [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    if (notes != null && notes.isNotEmpty) {
+      return StringList(
+          items: notes, padding: const EdgeInsets.fromLTRB(8, 0, 0, 0));
+    }
+    return Container();
+  }
+
+  Widget getHeading() {
```
Comment (maks) [inline lib/widget/lists/ventilation_guide_card.dart]: private pls

File: lib/widget/lists/ventilation_guide_card.dart [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+          Text(
+            sectionItem.heading,
+            style: Styles.textH4,
+          ),
+          const EdgeInsets.fromLTRB(16, 24, 16, 16));
+    }
+    return Container();
+  }
+
+  Widget getPaddedItem(Widget child, EdgeInsets padding) {
```
Comment (maks) [inline lib/widget/lists/ventilation_guide_card.dart]: private pls

File: lib/widget/lists/ventilation_guide_card.dart [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+                            mainAxisSize: MainAxisSize.min,
+                            children: <Widget>[
+                              Text(item.title, style: Styles.textH4),
+                              getList(item.notes)
+                            ],
+                          ),
+                        ),
+                      )
+                    ],
+                  )))
```
Comment (maks) [inline lib/widget/lists/ventilation_guide_card.dart]: trailing commas pls

File: lib/widget/lists/ventilation_guide_card.dart [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return ReusableCardBase(
+        elevation: 0,
+        color: backgroundColor,
+        child: <Widget>[
+          getHeading(),
+          ...sectionItem.list
+              .map((item) => Padding(
```
Comment (maks) [inline lib/widget/lists/ventilation_guide_card.dart]: would be nice to break this (from Padding on down) out into a separate widget or at least a private "builder" function that takes an `item`

**Review by maks — CHANGES_REQUESTED** [review]
:100: :+1: excellent work @ericjiang97 ! Just a few small nits from me.

File: lib/models/ventilation_guide.dart (hunk 1 above)
Comment (ericjiang97) [reply lib/models/ventilation_guide.dart:1]: :)

**Review by maks — APPROVED** [review]
:tada:

**Comment by lukesleeman** [comment]
One thing I noticed is the PDF button leads nowhere - I'll fire up a new issue so we can track this in a different PR

File: lib/view/airway/guidance/intubation_guidance_page.dart [hunk 11]
```diff
@@ -6,42 +6,28 @@ import '../../../hard_data.dart';
...
-          iconTheme: Styles.appBarIconTheme,
-          bottom: TabBar(
-            labelColor: AppColors.blackAlpha800,
-            labelStyle: Styles.textSemiBold,
-            unselectedLabelColor: AppColors.tabBarDeselectedText,
-            indicatorColor: AppColors.green900,
-            tabs: intubationGuide.map((e) => Tab(text: e.name)).toList(),
-            isScrollable: true,
+    return TabViewTemplate(
+      'Intubation Checklist',
```
Comment (lukesleeman) [inline lib/view/airway/guidance/intubation_guidance_page.dart]: Just noticed this should "Intubation Guide"
```suggestion
      'Intubation Guide',
```

**Review by lukesleeman — COMMENTED** [review]
Just noticed that the title of a page got changed!

File: lib/view/airway/guidance/intubation_guidance_page.dart (hunk 11 above)
Comment (lukesleeman) [reply lib/view/airway/guidance/intubation_guidance_page.dart]: Also big hat tip to @sallychung for noticing this! 👍 👍 🙌

**Pull request merged**